From 82d99dad0e8182a2c22e498e76bf797a10a30c71 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 26 Nov 2024 11:11:53 +0100
Subject: [PATCH 756/841] clk: clk-stm32-core: add trace for unknown clock
 identifier

Add a trace to identify usage of unknown clock identifier with
clock API.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git aecb88d96c09acd1a589769cccf546a666c959fa clk: clk-stm32-core: add trace for unknown clock identifier]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Iea1f7de3e7692c6d27c714ed377c4c69a31d1e3d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/420768
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/st/clk/clk-stm32-core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/st/clk/clk-stm32-core.c b/drivers/st/clk/clk-stm32-core.c
index bc1e0aa7c..86d1eacf0 100644
--- a/drivers/st/clk/clk-stm32-core.c
+++ b/drivers/st/clk/clk-stm32-core.c
@@ -569,6 +569,7 @@ static int clk_stm32_enable(unsigned long binding_id)
 
 	id = clk_get_index(priv, binding_id);
 	if (id == -EINVAL) {
+		ERROR("%s: unsupported clock id %lu\n", __func__, binding_id);
 		return id;
 	}
 
@@ -583,6 +584,8 @@ static void clk_stm32_disable(unsigned long binding_id)
 	id = clk_get_index(priv, binding_id);
 	if (id != -EINVAL) {
 		_clk_stm32_disable(priv, id);
+	} else {
+		ERROR("%s: unsupported clock id %lu\n", __func__, binding_id);
 	}
 }
 
@@ -593,6 +596,7 @@ static bool clk_stm32_is_enabled(unsigned long binding_id)
 
 	id = clk_get_index(priv, binding_id);
 	if (id == -EINVAL) {
+		ERROR("%s: unsupported clock id %lu\n", __func__, binding_id);
 		return false;
 	}
 
@@ -606,6 +610,7 @@ static unsigned long clk_stm32_get_rate(unsigned long binding_id)
 
 	id = clk_get_index(priv, binding_id);
 	if (id == -EINVAL) {
+		ERROR("%s: unsupported clock id %lu\n", __func__, binding_id);
 		return 0UL;
 	}
 
@@ -619,6 +624,7 @@ static int clk_stm32_get_parent(unsigned long binding_id)
 
 	id = clk_get_index(priv, binding_id);
 	if (id == -EINVAL) {
+		ERROR("%s: unsupported clock id %lu\n", __func__, binding_id);
 		return id;
 	}
 
-- 
2.47.2

